From d90360a59bc7ce706650b54d09224121d7a30613 Mon Sep 17 00:00:00 2001
From: Nathan Chan <e334cn@gmail.com>
Date: Fri, 22 Nov 2013 12:31:38 -0800
Subject: [PATCH 1/1] #ifdef for ICS Audio Blob compatibility

see hardware/libhardware/include/hardware/audio.h, get_presentation_position is not defined when ICS_AUDIO_BLOB flag is enabled

Change-Id: Ib3d99e9637c5d8e1243c5d49d5bdb2c034bc2a41
---
 media/libnbaio/AudioStreamOutSink.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/media/libnbaio/AudioStreamOutSink.cpp b/media/libnbaio/AudioStreamOutSink.cpp
index d5937b2..d19396c 100644
--- a/media/libnbaio/AudioStreamOutSink.cpp
+++ b/media/libnbaio/AudioStreamOutSink.cpp
@@ -84,6 +84,7 @@ status_t AudioStreamOutSink::getNextWriteTimestamp(int64_t *timestamp) {
 
 status_t AudioStreamOutSink::getTimestamp(AudioTimestamp& timestamp)
 {
+#ifndef ICS_AUDIO_BLOB
     if (mStream->get_presentation_position == NULL) {
         return INVALID_OPERATION;
     }
@@ -95,6 +96,9 @@ status_t AudioStreamOutSink::getTimestamp(AudioTimestamp& timestamp)
     }
     timestamp.mPosition = position64;
     return OK;
+#else
+    return INVALID_OPERATION;
+#endif
 }
 
 }   // namespace android
-- 
1.8.3.2

